### DISCUSSION TITLE
Draft Meetings Do Not Export CSV

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#419] [Q&A] [answered]
None of my draft meetings export to the CSV file.

I put about 40 meetings in draft mode and did an export, but the only meetings that were on the spreadsheet were published ones. 

How can I download them?

**Comment by tech2serve** [comment, marked as answer]
I’m converting to Q&A Discussion.

**Reply by Grateful-Viv** [reply]
Okay, let's discuss. How can how download them?

**Reply by tech2serve** [reply]
TSML doesn't provide a way to do that since they aren't seen as a valid meeting members can attend. 

A quick Internet search indicates there may be some other plugins you can experiment with to export draft posts.

**Comment by Grateful-Viv** [comment]
The problem is when importing the csv back into the database because it was easier to do a search and replace in Excel to fix some of the entries, it overwrote all the data and did not save any of the meetings in draft status.  For testing purposes, I put all inactive meetings in draft mode. 

There are 3 options to import 

1. don't delete anything - which just adds whatever is imported in and there could be many many duplicates.
2. delete only the meetings, locations, and groups for the regions present in this CSV - which I hoped would have preserved the draft files. Instead it overwrote everything and acted just like option 3 below.
3. delete all meetings, locations, groups, districts, and regions

The only fortunate thing is this was all on the staging site. But this is not good if I ever need to do a wholesale cleanup on the csv and also want to keep the draft files intact.

So my question is why doesn't TSML include draft meetings in the csv export?

**Reply by tech2serve** [reply]
I just tested with Import Option 2, and it didn't delete the draft meeting. I had minimal data in the meeting, but it persisted.

If this doesn't work for you, I see a couple of options:

1. Don't put your inactive meetings in draft status. Instead use the inactive options TSML provides. Or, export them separately to a file and add them back in as they become open again.
2. Don't export/import meetings in bulk. Instead use the edit meeting feature until such time as all your meetings have re-opened or you remove them.

**Reply by Grateful-Viv** [reply]
I will try one or more of your suggestions. Glad this wasn’t live data and glad to have back up redundancies. 

I’d use option 1 a lot more if the upcoming meeting widget could filter out inactive meetings. 

And this ties right back to the issue of the active filter via URL not working or staying on. Any idea when this will be addressed?

**Comment by Grateful-Viv** [comment]
Instead of putting the meetings in draft mode and leaving them there, I put them in the trash (and did not empty). So when I redid the import, the trash was untouched. Then I bulk restored them back into draft.

**Reply by tech2serve** [reply]
Excellent. 

If you don’t mind, please accept this or the answer above to close this out.

**Reply by Grateful-Viv** [reply]
This isn't an ideal resolution and most people will not know this.

**Reply by tech2serve** [reply]
This is not a widespread issue. It is occurring because you're trying to do something the plugin isn't designed to do. Hence, the workaround. 

It is documented here in discussions if anyone come searching.

I'm working on a change that filters out inactive meetings in the widget (#423). That will be the extent of my work on this.

**Reply by Grateful-Viv** [reply]
Putting them in the trash removed group info and location info, :(
